File: frontend/src/utils/videoModels.js
export const VIDEO_MODELS = [
  { id: "automatic_omni_mini", label: "Seedance 2.0 Mini — saved Automatic selection", family: "Seedance 2.0" },
  { id: "seedance_evolink", label: "Seedance 2.0 Standard — EvoLink", family: "Seedance 2.0" },
  { id: "seedance_fast_evolink", label: "Seedance 2.0 Fast — EvoLink", family: "Seedance 2.0" },
  { id: "seedance_mini_evolink", label: "Seedance 2.0 Mini — EvoLink", family: "Seedance 2.0" },
  { id: "seedance_fal", label: "Seedance 2.0 Standard — fal", family: "Seedance 2.0" },
  { id: "seedance_fast_fal", label: "Seedance 2.0 Fast — fal", family: "Seedance 2.0" },
  { id: "seedance_mini_fal", label: "Seedance 2.0 Mini — fal", family: "Seedance 2.0" },
  { id: "kling_voice_fal", label: "Kling 3 Voice ID — fal · 4K", family: "Kling 3.0" },
  { id: "kling_avatar_fal", label: "Kling Avatar — fal · Experimental", family: "Kling 3.0" },
];
export function modelNote(id) {
  if (id === "automatic_omni_mini") return "All scenes now use Seedance Mini via EvoLink, including silent scenes, dialogue and narration. This saved selection uses 720p.";
  if (id === "seedance_mini_evolink" || id === "seedance_mini_fal") return "Seedance Mini for every scene. Uses your accepted preview and character references, plus approved speech for dialogue or narration. Supports 480p and 720p.";
  if (id === "kling_voice_fal") return "All shots use Kling. English/Chinese speech only; always 4K. Generates new speech in the saved voice. First voice setup needs 5–30 seconds of approved speech and an extra setup call.";
  if (id === "kling_avatar_fal") return "Speaking shots only. Uses the preview and approved audio; scene motion, duration and resolution are model-controlled. Silent or narration-only shots cannot use this model.";
  return "All shots use this model. Speaking shots include the approved audio reference; review the generated words and timing. Fast and Mini support up to 720p.";
}

File: backend/app/video_models.py
"""Job-level rendering choices, independent of Compiler model-family names."""
from typing import Literal
AUTOMATIC = "automatic_omni_mini"  # Legacy saved-job ID; now Mini-only, never Omni.
VideoModel = Literal["h3_max_fal", "seedance_evolink", "seedance_fal", "seedance_fast_evolink", "seedance_mini_evolink",
                     "seedance_fast_fal", "seedance_mini_fal", "kling_voice_fal", "kling_avatar_fal", "automatic_omni_mini"]


def validate_selection(model, family, language, quality):
    if not model:
        return
    if model == AUTOMATIC and quality != "720p":
        raise ValueError("Automatic uses the verified 720p output tier; select 720p")
    expected = "MiniMax H3 Max" if model == "h3_max_fal" else ("Kling 3.0" if model.startswith("kling_") else "Seedance 2.0")
    if family != expected:
        raise ValueError("The selected video model must match its planning model family")
    if model == "kling_voice_fal" and language.strip().lower() not in {"english", "chinese", "en", "zh", "mandarin"}:
        raise ValueError("Kling voice-ID supports English/Chinese only. Choose Seedance for this language.")
    if ("_fast_" in model or "_mini_" in model) and quality not in {"480p", "720p"}:
        raise ValueError("Seedance Fast and Mini support 480p or 720p only")
